kubeconfig-creator/kubeconfig-creator.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kubeconfig-creator
+spec:
+  params:
+    - name: name
+      description: name of the cluster
+      type: string
+    - name: type
+      description: type of the resource
+      type: string
+      default: cluster
+    - name: url
+      description: address of the cluster
+      type: string
+    - name: username
+      description: username for basic authentication to the cluster
+      type: string
+    - name: password
+      description: password for basic authentication to the cluster
+      type: string
+      default: ""
+    - name: cadata
+      description: contains PEM-encoded certificate authority certificates
+      type: string
+      default: ""
+    - name: namespace
+      description: default namespace to use on unspecified requests
+      type: string
+      default: ""
+    - name: token
+      description: bearer token for authentication to the cluster
+      type: string
+      default: ""
+    - name: insecure
+      description: to indicate server should be accessed without verifying the TLS certificate
+      type: string
+      default: "false"
+  workspaces:
+    - name: kubeconfigFile
+  steps:
+    - name: write
+      image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/kubeconfigwriter:latest
+      command: ["/ko-app/kubeconfigwriter"]
+      args:
+        # passing the required json in the form of string to generate the kubeconfig file.
+        #
+        - -clusterConfig
+        - '{
+              "name":"$(params.name)",
+              "type":"$(params.type)",
+              "url":"$(params.url)",
+              "username":"$(params.username)",
+              "password":"$(params.password)",
+              "cadata":"$(params.cadata)",
+              "namespace":"$(params.namespace)",
+              "token":"$(params.token)",
+              "Insecure":$(params.insecure)
+            }'
+    - name: copy
+      image: quay.io/openshift/origin-cli:latest
```

Comment:
This image is 305 MB just for the `cp` and `cat` commands.  Could we use something smaller like an alpine or busybox?

Reply:
That was by mistake, i forgot to replace :sweat_smile: